download-maven-plugin/src/main/java/com/googlecode/WGet.java

```diff
@@ -215,12 +221,46 @@ public void execute() throws MojoExecutionException, MojoFailureException {
     // DO
     try
     {
-      if (outputFile.exists() && !overwrite)
+      boolean haveFile = outputFile.exists();
+      if (haveFile)
       {
+        if (this.checkSignature && this.md5 != null)
+        {
+          try {
+            SignatureUtils.verifySignature(outputFile, this.md5,
+                    MessageDigest.getInstance("MD5"));
+          } catch (MojoFailureException e) {
+            getLog().warn("File already exist. " + e.getMessage());
```

Comment:
This message doesn't seem very relevant here. I would suggest "The local cached version of file outputFile.getName() doesn't match the expected signature. You should consider checking the specified signature is correctly set, and/or cleaning your local cache."